### PR TITLE
adapt iets3.os to use snapshot repo

### LINKED ISSUE #823: Adapt iets3.opensource to use snapshot repo
Adapt the iets3.opensource gradle script to use the snapshot repository. 

- all except maintenance and cust. branches goes to SNAPSHOT.
- for an example see: [MPS-extensions/build.gradle L:83](https://github.com/JetBrains/MPS-extensions/blob/0f22380f75b8f53d08026a575c5b188c964b7a66/build.gradle#L83)

### PR DESCRIPTION
**Pull request by psuzzi**
All branches except main, maintenance and customer branches should go to the -SNAPSHOT branch. 

fix #823

### CODE REVIEW

**Comment by arimer** [comment]
@psuzzi I triggered your build manually on TC. Let's see if you see the artifact in the right location :)

**Comment by arimer** [comment]
According to the error message from TC, you would need to change the version of the artifact in question: 

```
Caused by: org.gradle.internal.resource.transport.http.HttpErrorStatusCodeException: 
Could not PUT 
'https://artifacts.itemis.cloud/repository/maven-mps-snapshots/org/iets3/opensource/fix-823_adapt_build_snapshot.2021.3.6851.21d7e42/opensource-fix-823_adapt_build_snapshot.2021.3.6851.21d7e42.zip'. Received status code 400 from server: Repository version policy: 
SNAPSHOT does not allow version: fix-823_adapt_build_snapshot.2021.3.6851.21d7e42
```

File: build.gradle [hunk 1]
```diff
@@ -82,10 +82,10 @@ if (project.hasProperty('iets3OpenSourceVersion')) {
         currentBranch = GitBasedVersioning.gitBranch
 
         def buildNumber = System.env.BUILD_NUMBER.toInteger()
-        if (currentBranch.startsWith("maintenance") || currentBranch.startsWith("mps")) {
+        if (currentBranch.startsWith("maintenance") || currentBranch.startsWith("mps") || currentBranch == "master" || currentBranch.contains("datev") ) {
```
Comment (sergej-koscejev) [inline build.gradle:85]:
```suggestion
        if (currentBranch.startsWith("maintenance/") || currentBranch.startsWith("datev/") || currentBranch.contains("/datev/")) {
```

Comment (sergej-koscejev) [reply build.gradle:85]: `currentBranch.contains("datev")` would treat `feature/datev/xyz` as a non-snapshot branch.

**Review by arimer — APPROVED** [review]
Thanks for updating the publication logic 👍🏼

**Pull request merged**